pkg/operator/ceph/csi/spec.go

```diff
@@ -428,3 +433,59 @@ func createCSIDriverInfo(clientset kubernetes.Interface, name string) error {
 
 	return err
 }
+
+// ValidateCSIVersion checks if the configured ceph-csi image is supported
+func ValidateCSIVersion(clientset kubernetes.Interface, namespace, csiPluginImage, rookImage, serviceAccountName string) error {
+	timeout := 15 * time.Minute
+
+	logger.Infof("detecting the ceph csi image version for image %s", csiPluginImage)
+
+	pods, err := clientset.CoreV1().Pods(namespace).List(metav1.ListOptions{LabelSelector: "app=" + operatorDeploymentName})
```

Comment:
There is a helper in `k8sutil.GetRunningPod()` that gets the current pod for you.